__tests__/app.test.js

```diff
@@ -254,23 +254,139 @@ describe("news api", () => {
         return request(app)
-          .get("/api/users")
-          .expect(200)
+          .post("/api/articles/1/comments")
+          .send(newPost)
+          .expect(400)
           .then(({ body }) => {
-            const usersArr = body.users;
-            expect(usersArr.length).toBe(4);
-            usersArr.forEach((user) =>
-              expect(
-                "username" in user && "name" in user && "avatar_url" in user
-              ).toBe(true)
+            const badRequest = body.msg;
+            expect(badRequest).toBe(
+              "Bad request. Request must be an object including username and body properties"
+            );
+          });
+      });
+      test("400: returns a 'Bad request. Request must be an object including username and body properties' message if request doesn't contain either of the properties", () => {
+        const newPost = {};
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send(newPost)
+          .expect(400)
+          .then(({ body }) => {
+            const badRequest = body.msg;
+            expect(badRequest).toBe(
+              "Bad request. Request must be an object including username and body properties"
+            );
+          });
+      });
+      test("400: returns a 'Bad request. Request must be an object including username and body properties' message if request isn't an object", () => {
+        const newPost = "invalid";
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send(newPost)
+          .expect(400)
+          .then(({ body }) => {
+            const badRequest = body.msg;
+            expect(badRequest).toBe(
+              "Bad request. Request must be an object including username and body properties"
             );
           });
       });
+      test("400: returns a 'Bad request. User is not registered' message if username is not found in the database", () => {
+        const newPost = { username: "randomguy", body: "test" };
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send(newPost)
+          .expect(400)
+          .then(({ body }) => {
+            const badRequest = body.msg;
+            expect(badRequest).toBe("Bad request. User is not registered");
+          });
+      });
+      test("404: returns a 'Bad path. Article with given id not found' message if article with given ID is not found in database", () => {
+        const newPost = { username: "butter_bridge", body: "test" };
+        return request(app)
+          .post("/api/articles/9999999/comments")
+          .send(newPost)
+          .expect(404)
+          .then(({ body }) => {
+            const badPath = body.msg;
+            expect(badPath).toBe("Bad path. Article with given id not found");
+          });
+      });
+      test("404: returns a 'Bad path. Article ID should be a number' message if article_id is not a number", () => {
```

Comment:
This would be 400 because the articleID is invalid. 404 not found works for a valid articleId which just isn't found in the db but 'banana' would never be an ID